src/libcore/convert.rs

```diff
@@ -10,15 +10,35 @@
 
 //! Traits for conversions between types.
 //!
-//! The traits in this module provide a general way to talk about
-//! conversions from one type to another. They follow the standard
-//! Rust conventions of `as`/`to`/`into`/`from`.
+//! The traits in this module provide a general way to talk about conversions from one type to
+//! another. They follow the standard Rust conventions of `as`/`to`/`into`/`from`.
+//!
+//! Like many traits, these are often used as bounds for generic functions, to support arguments of
+//! multiple types.
+//!
```

Comment:
I would add something along the following to the module documentation:

While essentially meaning the same thing, both `From` and `Into` exist because, due to the coherence rules, one or the other may not be implementable in certain situations. Because both basically mean the same thing, any conversion possible using `From` is also always possible using `Into` (although the reverse is not true). For this reason, if implementing `From` or `Into` for a custom type prefer `From`. If using `From` or `Into` in a type bound prefer `Into`.

(I'm not completely sure about the factual correctness of the above, but this is the situation as I currently understand it)